src/k8s/pkg/k8sd/app/hooks_start.go

```diff
@@ -14,13 +14,13 @@ import (
 	"github.com/canonical/microcluster/state"
 )
 
-func (a *App) onStart(s *state.State) error {
+func (a *App) onStart(ctx context.Context, s state.State) error {
```

Comment:
For the `OnStart` context, the full lifecyle context is passed. See: https://github.com/canonical/microcluster/pull/190